### ISSUE TITLE
Clear data log when main.py is opened for write

### ISSUE THREAD

**Issue by microbit-matt-hillsdon** [microbit-foundation/micropython-microbit-v2-simulator#48]
https://github.com/microbit-foundation/micropython-microbit-v2/commit/6ade36b3cb26f6b945606a61763422ed630155f5 introduced a change that clears the data log on opening main.py for write. We have our own main.py and our own file system so don't get that in the simulator for free.

However, we already clear the log on flash, so I think this would just catch self-rewriting programs in the simulator context.  For that reason I don't think it's worth doing right now as part of #47 as we're pressed for time.

**Comment by microbit-carlos** [comment]
I agree, if every time the "play" button is pressed it clears the log there is no need to do much else right now.

After launch we can look into adding the hook for self-modifying programmes, but it's safe to assume the impact will be almost insignificant.
